### PR TITLE
Fix numLedgersReplicated metric

### PR DESCRIPTION
**Pull request by wenbingshen**
### Motivation

When I want to add num not adhering placement ledgers replicated metric for RW(#3652), I found the numLedgersReplicated statistic to be inaccurate.

When we could not find a target bookie to replicate the ledger, the numLedgersReplicated metric will still increase.
image

### CODE REVIEW

**Comment by wenbingshen** [comment]
ping @zymap @hangc0276 If you have time. PTAL.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java [hunk 1]
```diff
@@ -1145,9 +1145,10 @@ public void replicateLedgerFragment(LedgerHandle lh, final LedgerFragment ledger
         if (MapUtils.isEmpty(targetBookieAddresses)) {
             LOG.warn("Could not replicate for {} ledger: {}, not find target bookie.",
                     ledgerFragment.getReplicateType(), ledgerFragment.getLedgerId());
-            return;
+            return false;
```
Comment (hangc0276) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java:1148]: I think we can directly throw a `BKException.BKLedgerRecoveryException` here.

**Comment by hangc0276** [comment]
@horizonzy  Please help take a look at this PR, thanks.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java (hunk 1 above)
Comment (horizonzy) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java:1148]: +1

Comment (wenbingshen) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java]: Addressed. PTAL

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java [hunk 2]
```diff
@@ -1262,6 +1267,9 @@ protected EnsemblePlacementPolicy initializeEnsemblePlacementPolicy(ClientConfig
             assertNull(stat1);
         });
 
+        assertEquals("NUM_FULL_OR_PARTIAL_LEDGERS_REPLICATED",
```
[the remark this answers is not in the document]
Comment (hangc0276) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java:1270]: @wenbingshen Would you please take a look at this comment? thanks a lot.

[the remark this answers is not in the document]
Comment (wenbingshen) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java]: @horizonzy @hangc0276 addressed done. PTAL

**Comment by wenbingshen** [comment]
> https://github.com/apache/bookkeeper/actions/runs/3675989711/jobs/6216085989 Check style failed.

@horizonzy Thanks for the reminder, I've fixed it.

**Review by hangc0276 — APPROVED** [review]

**Comment by hangc0276** [comment]
@zymap @horizonzy Would you please help take a look at this PR? thanks a lot.

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java [hunk 3]
```diff
@@ -1156,11 +1160,49 @@ public void testRWShutDownInTheCaseOfZKOperationFailures() throws Exception {
...
+    }
+
+    @Test
+    public void testReplicationStats() throws Exception {
+        BiConsumer<Boolean, ReplicationWorker> checkReplicationStats = (first, rw) -> {
+            try {
+                final Method rereplicate = rw.getClass().getDeclaredMethod("rereplicate");
+                rereplicate.setAccessible(true);
+                final Object[] objects = new Object[0];
+                final Object result = rereplicate.invoke(rw, objects);
```
Comment (horizonzy) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java:1173]: We can use `rereplicate.invoke(rw);` directly

Comment (wenbingshen) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java]: @horizonzy Thanks. I have addressed it.

**Review by horizonzy — APPROVED** [review]
LGTM.

**Pull request merged**